Automattic/wp-calypso
# Email: Surface different warning message when the email subscription and domain are owned by different users

#### Proposed Changes

In the case where the domain and email subscription are owned by different users, our [new messaging](https://github.com/Automattic/wp-calypso/pull/68435) that suggests the user to contact the owner of the email / domain will not work. Since the user must be both the domain and email subscription owner in order to make a purchase. Therefore, we will need separate messaging to address this scenario, and this PR introduces a new component for this case.

#### Testing Instructions


* Have two different user accounts - User A and User B
* Have a Professional Email subscription as User A
* Follow the steps here pbLl1t-LC-p2 to transfer the Professional Email subscription to User B
* As User A, visit Upgrades > Emails, you should see the new messaging introduced in this PR
* As User B, visit Upgrades > Emails, you should see the new messaging introduced in this PR

<img width="999" alt="Screen Shot 2022-10-04 at 2 43 24 PM" src="https://user-images.githubusercontent.com/104910361/193900555-59aee1c9-a603-4c8e-934a-35aaa315a12a.png">


#### Pre-merge Checklist






Related to #
